### PR TITLE
Rename method and reagents fields (#44)

### PR DESCRIPTION
**Pull request by danielsotirhos**

### CODE REVIEW

**Comment by danielsotirhos** [comment]
The bundle used for test_collection_protocol_fields() is a bit large, perhaps too large?
Also, the new test cases were added to match the existing test case structure, however not all bundles tested by _assert_bundle() contain the necessary fields to be tested (eg. CollectionProtocol.method) which is why testing of these fields was implemented by looping over bundle.protocols.values() and testing what was found in there. If a less forgiving implementation of these tests is desired then these checks will have to be pulled out of _assert_bundle() and put in the individual test cases.

File: test/test.py [hunk 1]
```diff
@@ -264,7 +315,7 @@ def _assert_bundle(self, uuid, version, manifest, metadata_files,
         self.assertEqual(storage_methods, {s.storage_method for s in bundle.specimens})
         self.assertEqual(preservation_methods, {s.preservation_method for s in bundle.specimens})
 
-        print(json.dumps(as_json(bundle), indent=4))
+        # print(json.dumps(as_json(bundle), indent=4))
```
Comment (hannes-ucsc) [inline test/test.py]: This does actually have a purpose but it wasn't clear.

```suggestion
        # Prove that as_json returns a valid JSON structure (no cycles, correct types, etc.)
        self.assertTrue(isinstance(json.dumps(as_json(bundle), str))
```

This change is unrelated, so it should go into a separate commit.

File: test/test.py [hunk 2]
```diff
@@ -133,31 +137,34 @@ def test_v5_bundle(self):
...
-
-    def test_vx_analysis_specimen_bundle(self):
-        """
-        A vx primary bundle with a specimen_from_organism as sequencing input
-        """
-        self._test_bundle(uuid='3e7c6f8e-334c-41fb-a1e5-ddd9fe70a0e2',
-                          version='2018-09-20T230221.622042Z',
-                          deployment='staging',
-                          diseases={'glioblastoma'}),
+    # TODO: Use bundle from production to fix test broken by missing bundle
```
Comment (hannes-ucsc) [inline test/test.py]: Use `@unittest.skip("Explain bla bla")`

File: test/test.py [hunk 3]
```diff
@@ -179,7 +194,32 @@ def test_preservation_storage_bundle(self):
                           diseases={'normal'},
                           project_roles={'Human Cell Atlas wrangler', None, 'external curator'},
                           storage_methods={'frozen, liquid nitrogen'},
-                          preservation_methods={'cryopreservation, other'})
+                          preservation_methods={'cryopreservation, other'},
+                          library_construction_methods={'Smart-seq2'})
+
+    def test_dissociation_protocol_fields(self):
```
Comment (hannes-ucsc) [inline test/test.py]: Do we need this test still, now that yu removed the changes to DissociationProtocol?

File: test/test.py [hunk 4]
```diff
@@ -179,7 +194,32 @@ def test_preservation_storage_bundle(self):
...
+        with dissociation_method and protocol_reagents fields
+        """
+        self._test_bundle(uuid='6b498499-c5b4-452f-9ff9-2318dbb86000',
+                          version='2019-01-03T163633.780215Z',
+                          age_range=AgeRange(1734480000.0, 1860624000.0),
+                          diseases={'normal'},
+                          project_roles={None, 'principal investigator', 'Human Cell Atlas wrangler'},
+                          library_construction_methods={"Chromium 3' Single Cell v2"})
+
+    def test_collection_protocol_fields(self):
```
Comment (hannes-ucsc) [inline test/test.py]: Same question as above.

File: test/test.py (hunk 3 above)
Comment (danielsotirhos) [reply test/test.py]: Correct, the tests for DissociationProtocol and CollectionProtocol fields are no longer needed. I've just committed change 4d15ecc to remove them.

**Comment by hannes-ucsc** [comment]
Thank you, @danielsotirhos. I'm not gonna cut a release yet because I think you're fairly close with the ontology changes and it would makes sense to include that as well in the next release. Once I cut a release, you need to bump the dependency in Azul.
